File: apps/flutter_app/lib/models/event.dart
enum EventType { wedding, owambe, naming, funeral, birthday, other }

enum EventPrivacy { public, private }

enum EventStatus { active, revealed, archived }

enum EventTier { free, standard, premium }

class Event {
  final String id;
  final String ownerUserId;
  final String name;
  final String location;
  final EventType eventType;
  final EventPrivacy privacy;
  final DateTime startsAt;
  final DateTime endsAt;
  final DateTime revealAt;
  final EventTier tier;
  final EventStatus status;
  final int guestCount;
  final int photoCount;
  final DateTime createdAt;

  Event({
    required this.id,
    required this.ownerUserId,
    required this.name,
    this.location = '',
    required this.eventType,
    required this.privacy,
    required this.startsAt,
    required this.endsAt,
    required this.revealAt,
    required this.tier,
    required this.status,
    required this.guestCount,
    required this.photoCount,
    required this.createdAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) => Event(
        id: json['id'] as String,
        ownerUserId: json['owner_user_id'] as String,
        name: json['name'] as String,
        location: (json['location'] as String?) ?? '',
        eventType: EventType.values.byName(json['event_type'] as String),
        privacy: EventPrivacy.values.byName(json['privacy'] as String),
        startsAt: DateTime.parse(json['starts_at'] as String),
        endsAt: DateTime.parse(json['ends_at'] as String),
        revealAt: DateTime.parse(json['reveal_at'] as String),
        tier: EventTier.values.byName(json['tier'] as String),
        status: EventStatus.values.byName(json['status'] as String),
        guestCount: (json['guest_count'] as int?) ?? 0,
        photoCount: (json['photo_count'] as int?) ?? 0,
        createdAt: DateTime.parse(json['created_at'] as String),
      );

  Map<String, dynamic> toJson() => {
        'name': name,
        'location': location,
        'event_type': eventType.name,
        'privacy': privacy.name,
        'starts_at': startsAt.toIso8601String(),
        'ends_at': endsAt.toIso8601String(),
        'reveal_at': revealAt.toIso8601String(),
        'tier': tier.name,
      };
}

File: apps/api/app/models/event.py
import uuid
from datetime import datetime

from sqlalchemy import String, DateTime, Enum as SAEnum
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base
from app.schemas import EventType, EventPrivacy, EventStatus, EventTier


class Event(Base):
    __tablename__ = "events"

    id: Mapped[str] = mapped_column(String, primary_key=True, default=lambda: uuid.uuid4().hex[:12])
    owner_user_id: Mapped[str] = mapped_column(String, index=True)
    name: Mapped[str] = mapped_column(String(255))
    # Free-text venue/city — kept loose for v1. Once we add maps we'll add
    # `location_lat/lng` alongside; this stays as the human-readable label.
    location: Mapped[str] = mapped_column(String(255), default="")
    event_type: Mapped[EventType] = mapped_column(SAEnum(EventType))
    privacy: Mapped[EventPrivacy] = mapped_column(SAEnum(EventPrivacy), default=EventPrivacy.public)
    starts_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
    ends_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
    reveal_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
    tier: Mapped[EventTier] = mapped_column(SAEnum(EventTier), default=EventTier.free)
    status: Mapped[EventStatus] = mapped_column(SAEnum(EventStatus), default=EventStatus.active)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=datetime.utcnow)

    guests = relationship("Guest", back_populates="event", lazy="dynamic")
    photos = relationship("Photo", back_populates="event", lazy="dynamic")
